proxy.go

```diff
@@ -348,10 +353,25 @@ func handleVersion(showVersion bool) {
 	}
 }
 
+func setThreads() {
+	// If GOMAXPROCS has not been set, restrict our thread usage
+	// so we don't grow many idle threads on large core count systems,
+	// which un-necessarily consume host PID space (and thus set an
+	// artificial max limit on the number of concurrent containers we can
+	// run)
+	if os.Getenv("GOMAXPROCS") == "" {
```

Comment:
Just a thought - I wonder if it's worth adding a log call showing this value? The complication being that you can only do that after `setupLogger()` has been called. But you could still call `setThreads()` and then add the log call for `GOMAXPROCS` later I guess.